tests/rptest/tests/follower_fetching_test.py

```diff
@@ -0,0 +1,148 @@
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+import os
+from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+
+from rptest.services.cluster import cluster
+from rptest.services.kafka_cli_consumer import KafkaCliConsumer
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.services.redpanda import SISettings
+from rptest.tests.prealloc_nodes import PreallocNodesTest
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.util import wait_for_local_storage_truncate
+
+
+class FollowerFetchingTest(PreallocNodesTest):
+    def __init__(self, test_context):
+        self.log_segment_size = 1024 * 1024
+        self.local_retention = 2 * self.log_segment_size
+        si_settings = SISettings(
+            test_context,
+            cloud_storage_max_connections=5,
+            log_segment_size=self.log_segment_size,
+            cloud_storage_enable_remote_read=True,
+            cloud_storage_enable_remote_write=True,
+        )
+        self.s3_bucket_name = si_settings.cloud_storage_bucket
+
+        super(FollowerFetchingTest, self).__init__(test_context=test_context,
+                                                   num_brokers=3,
+                                                   node_prealloc_count=1,
+                                                   extra_rp_conf={
+                                                       'enable_rack_awareness':
+                                                       True,
+                                                   },
+                                                   si_settings=si_settings)
+
+    def tearDown(self):
+        self.cloud_storage_client.empty_bucket(self.s3_bucket_name)
+        super().tearDown()
+
+    def setUp(self):
+        # Delay startup, so that the test case can configure redpanda
+        # based on test parameters before starting it.
+        pass
+
+    def produce(self, topic, bytes=10 * 1024 * 1024):
+        msg_size = 512
+        msg_cnt = int(bytes / msg_size)
+        producer = KgoVerifierProducer(self.test_context, self.redpanda, topic,
+                                       msg_size, msg_cnt,
+                                       self.preallocated_nodes)
+        producer.start()
+        producer.wait()
+        producer.free()
+
+    def get_fetch_bytes(self, node, topic):
+        metrics = self.redpanda.metrics(node)
+        total_bytes = 0
+        for family in metrics:
+            for sample in family.samples:
+                if sample.name == "vectorized_cluster_partition_bytes_fetched_total" and sample.labels[
+                        'topic'] == topic:
+                    total_bytes += sample.value
+        return total_bytes
+
+    def create_consumer(self, topic, rack=None):
```

Comment:
We should probably parameterize this test on client implementations to factor in any client side quirks? (franz vs kafka-java etc).

Reply:
Agree, i will do it in the follow up PR were i am going to add more extensive test coverage. This only contains a basic test proving that the feature is working.